Create a RecommendedBadge component

bobsilverberg opened the pull request:
Fixes #7957 

Note that I am using the existing "trophy" icon that is used for "Staff Pick", but this will be replaced by a newer icon. I am waiting to acquire that from @brassy-. Note also that the color of the text is from the InVision mockup at https://mozilla.invisionapp.com/share/BXQTIIC86ZN.

Note also that we do not have a SUMO article to link to yet, so currently the link just points to SUMO. I have opened https://github.com/mozilla/addons-frontend/issues/7961 as a follow-up to update the link once it is available.

Here's a screenshot of the component in storybook: 
![Screenshot 2019-05-02 13 56 47](https://user-images.githubusercontent.com/142755/57095990-2d4de880-6ce2-11e9-89fc-158a85ba9e35.png)

bobsilverberg commented:
Oops, somehow the test didn't get added. Let me try to find it!

bobsilverberg commented on src/ui/components/RecommendedBadge/styles.scss:
```diff
@@ -0,0 +1,24 @@
+
+.RecommendedBadge {
+  align-items: center;
+  color: #e25920;
```
We should probably use a variable for this. This specific color does not seem to exist in `photon-colors`, so we'll need to create our own. I wasn't sure if I should just create one in this file, or in a higher level `vars` file.

bobsilverberg commented:
After working on integrating this into the details page, I made a few tweaks to the styles. Here's a new screenshot of storybook at `large` screen size:

![Screenshot 2019-05-02 15 37 57](https://user-images.githubusercontent.com/142755/57102071-989eb700-6cf0-11e9-8190-c1f4cc0fe0bb.png)

and one at less than `large` screen size:

![Screenshot 2019-05-02 15 38 13](https://user-images.githubusercontent.com/142755/57102093-a5230f80-6cf0-11e9-865f-fce94f3b31c9.png)

willdurand commented on src/ui/components/RecommendedBadge/index.js:
```diff
@@ -0,0 +1,41 @@
+    <div className="RecommendedBadge">
+      <a
+        className="RecommendedBadge-link"
+        href="https://support.mozilla.org/"
```
Do we need UTM params on this link? /cc @jvillalobos

bobsilverberg replied:
Good question. I assume that will all be revealed when we get the final URL, but I should try to remember to check for UTM params.

bobsilverberg replied:
Already done! :-) https://github.com/mozilla/addons-frontend/issues/7961

willdurand commented on src/ui/components/RecommendedBadge/styles.scss:
```diff
@@ -0,0 +1,33 @@
+@import '~ui/css/styles';
+
+$recommended-text-color: #e25920;
```
Is this part of Photon? https://design.firefox.com/photon/visuals/color.html

If not, shouldn't we be using a Photon color here?

bobsilverberg replied:
I added a comment about this to the diff when submitting this PR. The color specified in the InVision markup doesn't exist in `photon-colors`. There is an [orange color for Photon](https://design.firefox.com/photon/visuals/color.html#orange), but the style guide specifically states "Firefox Orange is only used for branding. Do not use it otherwise." so I'm not sure we should use that. @brassy- can you please comment on this?

bobsilverberg replied:
I guess we can use variations of [Firefox Orange](https://design.firefox.com/photon/visuals/color.html#orange) for now and await confirmation from @brassy- .

willdurand commented on src/ui/components/RecommendedBadge/styles.scss:
```diff
@@ -0,0 +1,33 @@
+
+  a:active,
+  a:hover {
+    color: $link-color;
```
Honestly, I find this color a bit ugly when used next to the Trophy icon color:

![Screen Shot 2019-05-03 at 10 10 08](https://user-images.githubusercontent.com/217628/57126100-fe8c4c80-6d8b-11e9-9cdd-ef163924bfd6.png)

bobsilverberg replied:
It's true, we should probably make this a different shade of orange, so that it's obvious to a user that it's changing, but not so ugly/jarring. @brassy- could you please comment on this as well?

willdurand commented on stories/ui/RecommendedBadge.js:
```diff
@@ -0,0 +1,15 @@
+      <Provider story={story()} />
+    </div>
+  ))
+  .add('default', () => <RecommendedBadge />);
```
I don't know if that's feasible but a RTL story would be nice.

bobsilverberg replied:
We can do that! https://www.npmjs.com/package/storybook-addon-rtl

willdurand commented on stories/ui/RecommendedBadge.js:
```diff
@@ -0,0 +1,15 @@
+import Provider from '../setup/Provider';
+
+storiesOf('RecommendedBadge', module)
+  .addDecorator((story) => (
```
Is this decorator needed?

bobsilverberg replied:
I was using it to provide an instance of `i18n`, but I guess I can just do that directly. I have updated the patch to do that.

willdurand commented:
> Note also that we do not have a SUMO article to link to yet, so currently the link just points to SUMO. I have opened #7961 as a follow-up to update the link once it is available.

Do we want to add `_target=blank`?

bobsilverberg commented:
Thanks for the review @willdurand. I have addressed your comments about the storybook story, but for all of the other ones we are awaiting more information.

bobsilverberg commented:
> > Note also that we do not have a SUMO article to link to yet, so currently the link just points to SUMO. I have opened #7961 as a follow-up to update the link once it is available.
> 
> Do we want to add `_target=blank`?

Yes, I think so. I've added it.

bobsilverberg commented:
I received the proper icon, and I have attempted to style it according to the [latest prototype](https://github.com/mozilla/addons-frontend/issues/7957#issuecomment-489238294). I have not been entirely successful, but it's a start and maybe someone can help me improve it.

Here's what it currently looks like in storybook:

<img width="174" alt="Screenshot 2019-05-03 16 58 09" src="https://user-images.githubusercontent.com/142755/57165474-32379880-6dc5-11e9-9520-c5ed46baa0e5.png">

bobsilverberg commented:
@MeridelW Can you confirm what the text should be when hovering over the recommended badge? I've looked through some of the docs about this, but I'm not sure of the final decision. I currently have "Recommended extensions are safe, high-quality extensions."

bobsilverberg commented on src/ui/components/RecommendedBadge/index.js:
```diff
@@ -0,0 +1,47 @@
+        rel="noopener noreferrer"
+        target="_blank"
+        title={i18n.gettext(
+          'Recommended extensions are safe, high-quality extensions.',
```
I asked @MeridelW in the issue to confirm whether that is the final text or not. I will open up and issue to track that question, as I gather we might land this before we have a final answer.

bobsilverberg commented on src/ui/components/RecommendedBadge/index.js:
```diff
@@ -0,0 +1,47 @@
+        className="RecommendedBadge-link"
+        href="https://support.mozilla.org/"
+        rel="noopener noreferrer"
+        target="_blank"
```
I'm not sure I understand. Do you mean add a test to assert that the prop is there?

willdurand replied:
No sorry, I meant manually looking at the generated HTML :p

bobsilverberg replied:
Looking at it in Storybook, I see `rel="noopener noreferrer"` on the `a` tag.

bobsilverberg commented:
> Nice! I played with the CSS and here is a patch: [gist.github.com/willdurand/15fe2ddba5aaec10198d0fd634a41584](https://gist.github.com/willdurand/15fe2ddba5aaec10198d0fd634a41584)
> It's not perfect but it goes in the direction that I'd like to take in order to merge this patch :)
> 
> ![2019-05-06 19 39 29](https://user-images.githubusercontent.com/217628/57243553-b80e4a80-7036-11e9-9db0-906c33008569.gif)

Thanks @willdurand, that is much better indeed. I noticed that it kind of breaks in rtl mode though:

<img width="194" alt="Screenshot 2019-05-06 14 00 14" src="https://user-images.githubusercontent.com/142755/57244832-4cf94f80-7007-11e9-931f-ecfc3bd3e4bb.png">

But I think I fixed that by using `padding-end` and `padding-start`.

This is ready for another look.

willdurand commented on src/ui/components/RecommendedBadge/styles.scss:
```diff
@@ -0,0 +1,57 @@
+@import '~ui/css/styles';
+
+$recommended-text-color: $orange-60;
+$recommended-hover-color: $orange-50;
```
Can we go for a darker color? I don't really see the difference..

bobsilverberg replied:
I suppose so. I doubt these will be the final colors. I am still waiting to hear back from @brassy- about which colors to use. FTR, I found the difference pretty obvious when hovering over the lint, but I'll change it.
